### ISSUE TITLE
Can you please bump this to a supported version of Electron?

### ISSUE THREAD

**Issue by axfelix** [0rpc/zerorpc-node#108]
Trying to rebuild zeromq is hell on non-Linux and now that Electron 2.x and 3.x are officially unsupported, using this module (which the whole Python Electron ecosystem depends on) is getting very painful.

**Comment by axfelix** [comment]
Zeromq is also finally bumping: https://github.com/zeromq/zeromq.js/issues/330

**Comment by bombela** [comment]
Can you make a pull request? After that I can release a new version.

**Comment by axfelix** [comment]
Yeah, I'll try to get to it sometime later this week, thanks!

**Comment by o1xhack** [comment]
Hi is there any update on the bump new electron version?

**Comment by axfelix** [comment]
Yeah, I was finding that the new zeromq wasn't actually reporting being compiled against a new Electron and I got stuck, haven't returned to it yet, you're welcome to give it a shot: https://github.com/zeromq/zeromq.js/issues/330

**Comment by o1xhack** [comment]
Previously I was using the 
* "zeromq": "5.1.1",
* "zerorpc": "^0.9.8"
* "electron": "6.0.11"
in my package.json for npm install. and under python2 environment it seems worked fine. Are you using py3?

**Comment by axfelix** [comment]
Yeah, I'm Python 3, but it's good to hear it's working for you. Are you doing `electron-rebuild` at all? There are some files I'm unfamiliar with that are required to rebuild zeromq with `.a` extensions that always break that workflow on Windows, so I've been stuck on using the precompiled libraries.

**Comment by o1xhack** [comment]
Yes, definitely `electron-rebuild`. The reason I said is python 3 is not work for me too.
I checked and seems the stable zeromq version v5 is not compatibility for py3, only work py2.

**Comment by axfelix** [comment]
Well, this is still highly of interest to me because right now I haven't been able to get a working environment with both a currently-supported Electron and a currently-supported Python, so I'm glad to know there are others interested in it.

**Comment by o1xhack** [comment]
I'm with Intelliji and created the virtual env based on python2.
Then using 
```
"zeromq": "5.1.1",
"zerorpc": "^0.9.8"
"electron": "6.0.11"
```
as I states cause this seems been the compatible version for both node+python

**Comment by axfelix** [comment]
I haven't written Python 2 code in years now and it goes unsupported in a matter of weeks, so that's not a great solution, though...

**Comment by o1xhack** [comment]
YES definitely not.
As I found out on zeromq github, they are testing the v6 beta that support higher Node and python3.
I failed on using that cause the zerorpc is still depend on a lower zeromq thus failed on electron-rebuild.

**Comment by abulka** [comment]
Sorry for jumping in, but several days of research have led me to this thread. The latest zerorpc 0.9.8 installs zeromq 4.6.0, which is from 2 years ago, and that version of zeromq will not rebuild via `$(npm bin)/electron-rebuild` which means you cannot use zerorpc with recent versions of electron.

Interestingly, when installed alone (using npm, without zerorpc), zeromq defaults to a much later 6.0.0-beta.5 version which can be rebuilt OK with electron-rebuild in the latest electron 7 (node 12 internally). But as soon as you `npm install zerorpc` the very old version of zeromq gets installed inside zerorpc (within node_modules), and fails to electron-rebuild. Only by downgrading to electron 3 can zerorpc be electron-rebuilt correctly. 

I thought this was an electron-builder problem and documented a really simple 4 step repro case electron/electron-rebuild#343 but was told it was not an electron builder problem. I'm starting to think it is a zerorpc problem, and that this thread is now the epicentre of an issue that has generated over a dozen unresolved forum posts and is affecting many devs. Am I on the right track?

**Comment by axfelix** [comment]
Correct and welcome 🙂

This is in fact why I'm still using electron 3 for my apps.

**Comment by o1xhack** [comment]
Just let you guys know, here with Electron 6.1.5 and zerorpc 0.9.8 works well in Mac

**Comment by axfelix** [comment]
Yeah, the issue is that electron-rebuild doesn't seem able to work on zeromq on Windows, so if you're supporting a cross-platform app, the only solution is bumping zerorpc to require newer zeromq builds, which is what I intended to do, but the new builds aren't declaring compatibility with new electron the way they're supposed to.

**Comment by o1xhack** [comment]
ok, I see, so electron 3 in win works fine with current zerorpc & rebuild?

**Comment by abulka** [comment]
@o1xhack Yes, just built a new release of my [Print42 app](https://print42.atug.com/) for windows yesterday with that combo - no probs.

Yesterday I emailed the zeromq-dev@lists.zeromq.org letting them know about this thread and asking for assistance - you never know, perhaps the gurus of zeromq might help out.

**Comment by bombela** [comment]
Does zeromq 6 beta fixes most of the problems? zeromq 6.0.0-beta.6 just came out. I wonder if it is reasonable to depend on the beta version.

**Comment by axfelix** [comment]
Unfortunately, it still doesn't ship prebuilt libraries that are compatible with a supported Electron, and it doesn't compile properly on Windows with electron-rebuild.

@rolftimmermans, are you aware of this on the zeromq side? It would be trivial to add a PR to zerorpc that requires the newest zeromq, (see https://github.com/axfelix/zerorpc-node/commit/0539fb169299505e95942c49281b76fe75e2e6d9), but I can't get a good build. I wonder if just merging this commit into this repo would be worthwhile anyway, since right now it won't install anything close to a supported electron version from upstream...

**Comment by jordanlevy96** [comment]
Hi is there any update on this?
